### ISSUE TITLE
iPad 2/IPad Retina crash on splash. 

### ISSUE THREAD

**Issue by m8tto** [robovm/robovm#1132]
Hello.
I'm using libgdx, And have app that perfectly works on real iPhone 5s, and in most simulator (except iPad 2/ipad retina launched with x64_86). When I deploy app via testflight it works on iPhone 5s, but doesn't work on iPads. 

I'm using several bindings (AdColony, AdMob(Robopod), GameCenter), but I excluded all of them, and the problem still there.

Also I re-setuped my Libgdx project to use latest libs. 
Using robovm gradle 1.6.1 SNAPSHOT.(in eclipse, but in build.grade 1.6.0 version, cause 1.6.1 can't download for some reason).

Sorry for any inconvenience.
Could you please help?

**Comment by badlogic** [comment]
Those two simulators don't support x86_64, but only x86. We fixed this im
the IDE plugins after the last release. For 1.6, you'll need to select x86
manually in the run configuration.

**Comment by m8tto** [comment]
No. I meant real devices. I understood about simulators. Will it work in release if testflight app crashes on em?

**Comment by badlogic** [comment]
We'll need more info, see http://docs.robovm.com/errors.html

**Comment by m8tto** [comment]
Did it.

Here is the log from device:

Incident Identifier: 1173B4BB-7255-4377-BE02-D821416B2ABD
CrashReporter Key:   7519cd39796ca6b76e0aa896bb95bf8926ba5542
Hardware Model:      iPad3,6
OS Version:          iPhone OS 8.4.1 (12H321)
Kernel Version:      Darwin Kernel Version 14.0.0: Wed Aug  5 19:24:36 PDT 2015; root:xnu-2784.40.6~18/RELEASE_ARM_S5L8955X
Date:                2015-08-26 23:38:21 +0300
Time since snapshot: 290 ms

Free pages:                              1722
Active pages:                            7402
Inactive pages:                          3924
Speculative pages:                       106
Throttled pages:                         202591
Purgeable pages:                         0
Wired pages:                             37020
File-backed pages:                       11008
Anonymous pages:                         424
Compressions:                            191307
Decompressions:                          3056
Compressor Size:                         35
Uncompressed Pages in Compressor:        68
Page Size:                               4096
Largest process:   IOSLauncher

Processes
     Name       |            <UUID>                |     CPU Time|     rpages|       purgeable| recent_max| lifetime_max| fds |  [reason]         | (state)

            adid <0caed189b8b937d9a1e8e91a6a9b1988>         0.062         108                0           -           586   50   [vm-pageshortage]   (daemon) (idle)
        misagent <1b94e4aadc6136baa8dda0fc5be9e2ee>         0.019          97                0           -           163   50   [vm-pageshortage]   (daemon) (idle)
   nsurlsessiond <266e2a4d17c43869825533cf27fbb01b>         0.708         353                0           -          1153   50   [vm-pageshortage]   (daemon)
mobile_assertion <24110c19e1b93c0db31510f2c40d718f>         0.032         112                0           -           429   50   [vm-pageshortage]   (daemon)
      MobileMail <69b8b84d5b4837a4810595141eeedb6f>         1.272         975                0           -          2971   50   [vm-pageshortage]   (resume) (continuous)
             lsd <925a795ee65236d88d04f1e13c2048c4>         0.114         174                0         +21           798   50   [vm-pageshortage]   (daemon)
            tccd <f60d6ade1e9030c4b480edad0fbbd999>         0.060         166                0           -           641   50   [vm-pageshortage]   (daemon)
             kbd <46569681dca8350dae6b272ef75d8f63>         0.222         271                0           -          1290   50   [vm-pageshortage]   (daemon)
     IOSLauncher <543ada7ba98b303fb65d4eb0b8fcc9d9>         4.983      158054                0           -        154144   50   [vm-pageshortage]   (audio) (frontmost) (resume)
      TestFlight <915b15d33c0236939b4b51866813efff>         1.130        2850                0           -          4046   50                       (frontmost) (resume)
            ptpd <d8fbb7a1322e30eaa38981c8ba210b31>         0.685         663                0           -          1641   50                       (daemon)
        BTServer <063882867dd43a0cb0b2e2d9f05c72f7>        11.951         485                0           -          1709   50                       (daemon)
       lockdownd <9fc5bc90e1fb3cdaaa4b55732bc70d0c>         7.487         290                0           -           907   50                       (daemon)
         imagent <01127b7b31e43fa7913558f2ea4d9b3a>         6.425         391                0           -          1097   50                       (daemon)
identityservices <6c3e0c39d866378b8174731f59e37585>        13.102         622                0           -          1693   50                       (daemon)
       locationd <b0b4f99ade98379ba7bc729ccee7f476>       261.596        1738                0           -          4821   50                       (daemon)
           wifid <da973a328bf23ce0a43b0aa2a400ef20>        53.479         526                0           -          1489   50                       (daemon)
      discoveryd <abf5bc1a20fc3a7986da8c53baecb945>        55.363         664                0           -          1389  100                       (daemon)
   iaptransportd <035b49c5fd32335389f5a93fe324904b>         4.642         322                0           -          1037   50                       (daemon)
    mediaserverd <314d1b0ac82b3132a865d1e0eaf1826d>      1324.585        1626                0           -         31105  200                       (daemon)
         syslogd <6f77eaad05f738bfae6899812e1d3860>        42.099         199                0           -           391   50                       (daemon)
        cfprefsd <0ea3288432a03a278ef3008f97bc758b>        18.791         332                0           -           613   50                       (daemon)
          powerd <0c49e68feda33c5d8018f0619d6f37d4>        17.951         245                0           -           552   50                       (daemon)
            apsd <97c9d1ea17043ab99c64b1e513991a33>        30.057         603                0           -          1743   50                       (daemon)
        networkd <28afec48070f34778f11462856a8a57a>        76.605        2253                0           -          2715   50                       (daemon)
        sharingd <2d59e562946d358795b1f87c86e85285>         9.854         666                0           -          1845   50                       (daemon)
     dataaccessd <91a0f1d9b8cf33a08fbe1cf83099f17c>        19.477        1267                0           -          2580   50                       (daemon)
         searchd <458cfad470213db8bac5ab67a12230bf>         0.965         798                0           -          2546   50                       (daemon)
    syslog_relay <36b9bb3e487b35c4a5cbfbfa87eeb612>         0.166         102                0           -           176   50                       (daemon)
    mediaremoted <d59a3bbc676d3cb386536624644310f3>         0.298         348                0           -          1321   50                       (daemon)
    itunesstored <59fa84b845aa39f699871ed7c62c1496>         1.627        1675                0           -          3967   50                       (daemon)
     SpringBoard <6b8bdeb54e7e3c808ce6d425d789fdb2>       523.320       11661                0           -         54064   50                      
      backboardd <919baf158ccc3ed1bef38a3e9f8c35ea>      3059.749       16794                0           -          5076   50                       (daemon)
  UserEventAgent <eb46a743d69e3829981e006aca302364>       170.116         923                0           -          1928  100                       (daemon)
         configd <3d2a55c1d9df385ab77ee83a6006dec9>        29.394         421                0           -          1203   50                       (daemon)
    fairplayd.A2 <dc27646d4dc73e75929f9ec8900d7c38>         9.996         162                0           -          1438   50                       (daemon)
       fseventsd <a489570c239132fe97d265b54f39c6a6>         8.087         342                0           -           830   50                       (daemon)
      assertiond <8c8271a8377032a8acebec3ae0449597>        18.420         312                0           -           778   50                       (daemon)
       distnoted <54a01364a6c1318bb01108a71025e6f1>         2.200         176                0           -           259   50                       (daemon)
   wirelessproxd <fa8e6480731b3e92ba4ab357cb204478>         3.799         196                0           -           728   50                       (daemon)
filecoordination <ccc8bb4aa45634ba9a46106b754cd0d4>         2.834         256                0           -           805  100                       (daemon)
      aggregated <482e2b8bb8a33d108554efb14bd1ee09>       626.185        1007                0           -          2195   50                       (daemon)
             ubd <f41764abcaf13d19958747f1ba57351e>         0.043         193                0           -           701   50                       (daemon)
   nsurlstoraged <e3abb27545b53bf7afecd5832558fb7e>        16.183         534                0           -          2308  100                       (daemon)
      CommCenter <e580116bcc79312384ed141388ce9882>       283.877        1483                0           -          5924   50                       (daemon)
         notifyd <4789659048203dcf943898f12c727462>        28.174         298                0           -           356   50                       (daemon)

**End**

**Comment by m8tto** [comment]
Is anything else needed?

**Comment by badlogic** [comment]
Sorry, it's 11pm over here, and i'm almost asleep :)

The log doesn't tell us much. Do you also happen to have a crash log? How
much work do you do before didFinishLaunching is called?

**Comment by m8tto** [comment]
You're not far away) (almost 0:00 here)

Well actually my IOSLauncher extends IOSApplication.Delegate and I do all initialization after createApplication method (which as far as i know launches after didFinishLaunching) so the answer is not much if any.

**Comment by badlogic** [comment]
Won't be able to diagnose this without a crash log then i'm afraid.

**Comment by m8tto** [comment]
How can I get one?
(I have no robovm license).
Or you mean device one?

**Comment by badlogic** [comment]
Crash log from the device will do, the docs describe how to get it.

**Comment by m8tto** [comment]
For now there's the only one that I've displayed. If it will be one, I'll post it here. Thanks. Good night)

**Comment by m8tto** [comment]
Nope. this type of log appear every time it crashes. Here is another one from the other iPad


Incident Identifier: 45D92B67-982E-4E5D-B0C9-4D70B887AC06
CrashReporter Key:   838ecfbc9ab9185d593c4e67972355e52f27ed47
Hardware Model:      iPad4,5
OS Version:          iPhone OS 8.4.1 (12H321)
Kernel Version:      Darwin Kernel Version 14.0.0: Wed Aug  5 19:25:56 PDT 2015; root:xnu-2784.40.6~18/RELEASE_ARM64_S5L8960X
Date:                2015-08-27 10:11:50 +0300
Time since snapshot: 219 ms

Free pages:                              9223
Active pages:                            108860
Inactive pages:                          48855
Speculative pages:                       7666
Throttled pages:                         0
Purgeable pages:                         2
Wired pages:                             49602
File-backed pages:                       17407
Anonymous pages:                         147974
Compressions:                            473098
Decompressions:                          150390
Compressor Size:                         32194
Uncompressed Pages in Compressor:        117149
Page Size:                               16384
Largest process:   IOSLauncher

Processes
     Name       |            <UUID>                |     CPU Time|     rpages|       purgeable| recent_max| lifetime_max| fds |  [reason]         | (state)

WirelessRadioMan <b5ed8bc92577333bb0bde4a7d0605700>         0.117         285                0           -           870   50                       (daemon) (idle)
coresymbolicatio <bb22d900394c38f899677a2dcc32d9d3>         0.010         106                0           -           221   50                       (daemon) (idle)
     diagnosticd <b8b88ecf403a3bea86f7e12d0fd28393>         0.008         105                0           -           213   50                       (daemon) (idle)
    itunescloudd <3260b558418b34eba40d3dda16c03c30>         0.284         732                0           -          2365   50                       (daemon)
   nsurlsessiond <6767a849847c39ae8e6334940d4e48b4>       110.893         867                0           -          2142   50                       (daemon)
            swcd <83ae947dad063d54985a37e0477d9312>         0.031         178                0           -           442   50                       (daemon)
            bird <52aae178219231b991ce9f7eb8fff6b0>         2.825         855                0           -          1839   50                       (daemon)
          cloudd <bb351ee716e73105bd3e9968eab0003d>        21.218        1345                0           -          3013   50                       (daemon)
         YouTube <b46b5af4c2f237d7ac344570afc3b119>       120.171       24248                0           -         25323  100                       (audio) (background)
        routined <7b2d141703713e71995ed8d7ec8f40c2>         0.009          20                0           -            53   50                       (daemon) (idle)
      MobileMail <7a67b896c6f73103a82396dcb793f01d>         6.558        1855                0           -          5586   50                       (resume) (continuous)
             lsd <eb0e256edfc435b791bdb90f8d3cc401>         2.131         424                0           -          1031   50                       (daemon)
            tccd <eda84621bba33c6985607830bbd6d671>         0.866         278                0           -           787   50                       (daemon)
             kbd <c23c08f51d1933f4ac9b4cc24e8a3587>         1.063        1061                0           -          2240   50                       (daemon)
       coreduetd <adafcd3f220c34a3aa195390e564c841>        26.811        1163                0           -          1966   50                       (daemon)
     IOSLauncher <1e9e0b1fda4c3f6386994f4af02a68b7>         3.375      182497                0           -        135343   50  [per-process-limit]  (audio) (frontmost) (resume)
      TestFlight <137c6b2b93b23294bacb0e03ae428d07>         2.302        4023                0           -          6067   50                       (frontmost) (resume)
        BTServer <e8393cbed5af397ea6c13032e66122d7>         1.099         509                0           -          1675   50                       (daemon)
           wifid <69af100cf4d138a2a2afe39e15031461>        20.697         619                0           -          1730   50                       (daemon)
       locationd <d828b7a53c7e343aac71a0ba9e865285>       102.413        2080                0           -          4595   50                       (daemon)
       lockdownd <7107c1be93dd343fa58a4b5e2af6cf15>         0.404         275                0           -           918   50                       (daemon)
      discoveryd <703eac07c6823551b351ca50c8617374>        36.382         685                0           -          1389  100                       (daemon)
identityservices <7d32337e76f73331b63504fd78403ec6>         5.077         873                0           -          2125   50                       (daemon)
         imagent <4ff9e2c4c33c3c6bac323e545febcc21>         2.245         508                0           -          1476   50                       (daemon)
          powerd <82fbbb9bbb8836b88b2b78d8ca022cc4>         7.356         250                0           -           505   50                       (daemon)
         syslogd <e985ebf1ab9638df9fbbd1af9565bf49>        10.674         230                0           -           332   50                       (daemon)
        cfprefsd <203df7aeb5613f2591a24bd33b373fdc>         7.616         404                0           -           757   50                       (daemon)
    mediaserverd <f9793a15ea9d3f1c916d436b30703dd4>      2082.776        8635                0           -         32257   50                       (daemon)
   iaptransportd <6687069506fc3ba6b6586d4cddff182a>         3.198         381                0           -          1419   50                       (daemon)
            apsd <e41cbfa9060e31a9a0963eb754a12b08>         6.463         693                0           -          1957   50                       (daemon)
        networkd <6d9c57d65d0635849e1bb8cadd6d6ad2>        51.287        1032                0           -          1712   50                       (daemon)
     dataaccessd <44721a09547a352da81b60440e015e0c>        10.364        1638                0           -          3964  100                       (daemon)
        sharingd <2a3948dd75db3615b1cdc8cd864fed73>         0.799         796                0           -          2519   50                       (daemon)
    mediaremoted <2d2845cb4a3e3313ac3382380a783014>         0.586         443                0           -          1749   50                       (daemon)
      assistantd <6f8913c840d13733a62374a88189996b>         1.827         976                0           -          2811   50                       (daemon)
     SpringBoard <1c83b21c962e3b5791cb518f829d9aa4>       151.197       12417                0           -         36539   50                      
      backboardd <684cf908be40359686c55864b395b108>      1913.396       20348                0           -          4760   50                       (daemon)
  UserEventAgent <e23056e8e53d3077af26829f9abdc867>       113.046         933                0           -          2428  200                       (daemon)
       fseventsd <1f3a77a5cb1f3b0d8b0724186fb0e3ac>         2.627         355                0           -           437   50                       (daemon)
         configd <5ca385b2232439d5bccd111e065f77b3>        16.685         548                0           -          1331   50                       (daemon)
    fairplayd.A2 <ea6d021cd6bc38158d25eef7d268b74a>         1.847         185                0           -          1596   50                       (daemon)
   wirelessproxd <41904877de153515bcce5dac1203f63b>         0.137         237                0           -           792   50                       (daemon)
      assertiond <04bfb60d7d963e37b769c9ad7a15e06b>         5.712         320                0           -           893   50                       (daemon)
       distnoted <829402ac5fbe37bbb323304d2424319a>         2.442         221                0           -           354   50                       (daemon)
filecoordination <f92a0133924039cf9c93b11126fe5b57>         0.530         276                0           -           858   50                       (daemon)
      aggregated <ff22d37c5925349da4364e4667cd5693>       261.175        1356                0           -          2642   50                       (daemon)
             ubd <72a7c79bedf53f079270d500f2a1ba87>         0.050         250                0           -           821   50                       (daemon)
   nsurlstoraged <a77939a4af6632e39c6ad798e8133f48>         5.278        1143                0           -          1489  100                       (daemon)
      CommCenter <c39a74c6e0a33ff7ae0862458f2307a0>        61.592        1556                0           -          4899   50                       (daemon)
         notifyd <3d474663ff4d3afdb2b658f31641ba45>        13.084         363                0           -           423   50                       (daemon)
     ReportCrash <1a209928bf773ebabad1e42630986f8e>         0.100         350                0           -           680   50                       (daemon)
     ReportCrash <1a209928bf773ebabad1e42630986f8e>         0.024         173                0           -           479   50                       (daemon)

**End**

**Comment by m8tto** [comment]
May be this line?
IOSLauncher <1e9e0b1fda4c3f6386994f4af02a68b7> 3.375 182497 0 - 135343 50 per-process-limit (frontmost) (resume)
Sounds like it uses resources over the limit

**Comment by badlogic** [comment]
You app is using to much memory i believe. You have 130k pages allocated,
one page is 4kb in size, meaning your app uses 528MB of memory. That's to
much for the iPad2. This includes all your textures and audio assets. Try
reducing that number.

**Comment by m8tto** [comment]
Hm. Only have 2 atlasses 1024 by 1024 several short sounds and one bg music. And one font. And the're not heavy. Looks like leak somewhere, and still iphone exp. Is perfect and fast

**Comment by badlogic** [comment]
Really hard to say anything more from our end without a repro case. I got
an iPad2, so i guess i could reproduce it if you send a reproduction case.
Could be that one of the bindings you use goes berserk?

**Comment by m8tto** [comment]
I can add youu to testflight, or you mean something else

**Comment by m8tto** [comment]
I tried without bindings so do think so

**Comment by badlogic** [comment]
Reproduction case: self contained project that reproduces the issue
reliably. I'm afraid i don't have the resources to debug your full app.

**Comment by m8tto** [comment]
You mean nearly empty project that still crashes?

**Comment by badlogic** [comment]
Yes.

**Comment by m8tto** [comment]
I've managed to figure out that it happens when my fonts are generating with FreeTypeFontGenerator...

**Comment by badlogic** [comment]
Need more info. How many glyphs do you generate, at what size. This is best
handled on the libGDX forums/IRC/tracker imo.

**Comment by m8tto** [comment]
Thank you for your help.
The problem indeed was in fonts, I've already fixed it yesterday, and app is on a review now!

I had 2 fonts and 1 of them was at half size of a screen, the problem was that I generated too many characters (needed only numbers) and in too large height (real half of the screen).

So I 've reduced size twice (then just scaled it, looks pretty) and removed unused characters.

Sorry for opening issue here, not robovm problem.
